test/parallel/test-buffer-fill.js

```diff
@@ -181,25 +182,30 @@ deepStrictEqualValues(genBuffer(4, [hexBufFill, 1, -1]), [0, 0, 0, 0]);
 
 
 // Check exceptions
-assert.throws(() => buf1.fill(0, -1), /^RangeError: Out of range index$/);
-assert.throws(() =>
-             buf1.fill(0, 0, buf1.length + 1),
-              /^RangeError: Out of range index$/);
-assert.throws(() => buf1.fill('', -1), /^RangeError: Out of range index$/);
-assert.throws(() =>
-             buf1.fill('', 0, buf1.length + 1),
-              /^RangeError: Out of range index$/);
-assert.throws(() =>
-             buf1.fill('a', 0, buf1.length, 'node rocks!'),
-              /^TypeError: Unknown encoding: node rocks!$/);
-assert.throws(() =>
-             buf1.fill('a', 0, 0, NaN),
-              /^TypeError: encoding must be a string$/);
-assert.throws(() =>
-             buf1.fill('a', 0, 0, null),
-              /^TypeError: encoding must be a string$/);
-assert.throws(() =>
-             buf1.fill('a', 0, 0, 'foo'), /^TypeError: Unknown encoding: foo$/);
+assert.throws(
+  () => buf1.fill(0, -1),
+  /^RangeError(?:\[ERR_INDEX_OUT_OF_RANGE\])?: Index out of range$/);
+assert.throws(
+  () => buf1.fill(0, 0, buf1.length + 1),
+  /^RangeError(?:\[ERR_INDEX_OUT_OF_RANGE\])?: Index out of range$/);
```

Comment:
For the cases where the regex must still be provided, it would be easiest to change later on if the regex was defined once and assigned then used multiple times... e.g.

```js
const errRegex = /^RangeError .../;
assert.throws(() => { /* ... */ }, errRegex);
assert.throws(() => { /* ... */ }, errRegex);
```